File: tests/fixtures/sample.m
// Sample Objective-C module for Tree-sitter testing.
#import <Foundation/Foundation.h>

@protocol Greeter
- (NSString *)greet:(NSString *)name;
@end

NSString* greet(NSString* name) {
    return [NSString stringWithFormat:@"Hello, %@!", name];
}

NSString* farewell(NSString* name) {
    return [NSString stringWithFormat:@"Goodbye, %@!", name];
}

@interface Calculator : NSObject <Greeter>
- (NSInteger)add:(NSInteger)a to:(NSInteger)b;
- (NSInteger)multiply:(NSInteger)a by:(NSInteger)b;
@end

@implementation Calculator

- (NSString *)greet:(NSString *)name {
    return [NSString stringWithFormat:@"Calculator says hello, %@!", name];
}

- (NSInteger)add:(NSInteger)a to:(NSInteger)b {
    return a + b;
}

- (NSInteger)multiply:(NSInteger)a by:(NSInteger)b {
    return a * b;
}

@end

int main(void) {
    @autoreleasepool {
        NSString* msg = greet(@"world");
        NSString* bye = farewell(@"world");
        Calculator* calc = [[Calculator alloc] init];
        NSInteger result = [calc add:2 to:3];
        NSInteger product = [calc multiply:result by:2];
        NSLog(@"%@", msg);
    }
    return 0;
}

File: tests/fixtures/sample.swift
// Sample Swift module for Tree-sitter testing.
import Foundation

protocol Greeter {
    func greet(name: String) -> String
}

func greet(name: String) -> String {
    return "Hello, \(name)!"
}

func farewell(name: String) -> String {
    return "Goodbye, \(name)!"
}

class Calculator: Greeter {
    private var history: [Int] = []

    func greet(name: String) -> String {
        return "Calculator says hello, \(name)!"
    }

    func add(a: Int, b: Int) -> Int {
        let result = a + b
        history.append(result)
        return result
    }

    func multiply(a: Int, b: Int) -> Int {
        return a * b
    }
}

func main() {
    let msg = greet(name: "world")
    let bye = farewell(name: "world")
    let calc = Calculator()
    let result = calc.add(a: 2, b: 3)
    let product = calc.multiply(a: result, b: 2)
    print(msg)
}

main()
